### DISCUSSION TITLE
Arduino/Pico SerialUSB class inheritance - What am I doing wrong?

### DISCUSSION THREAD

**Discussion by Zitt** [earlephilhower/arduino-pico#599] [Q&A] [answered]
I'm trying to inherit functionality of the base SerialUSB class on a Pico in the arduino framework. I've created a header file which contains:

MySerial.h:
`#include <Arduino.h>

class MySerialUSB: public SerialUSB { 
};
extern MySerialUSB MySerial;
`

in my main ino:
`#include "MySerial.h"
`

upon trying to compile; I get a compile time error:
`MySerial.h:9:37: error: expected class-name before '{' token
 class MySerialUSB: public SerialUSB {
                                     ^`

I've tried including several different files but can't seem to get the compiler to pass. I've done this before on the Nano and haven't been unsuccessful. I'm sure I'm doing something simple but can't seem to figure it out.

**Comment by Andy2No** [comment, marked as answer]
Did you try including SerialUSB.h, either near the top of your .h file after #include <Arduino.h>, or before #include "MySerial.h" in your main .ino?

**Reply by earlephilhower** [reply]
Quick test works when you include the proper header file.  Thanks, @Andy2No

**Reply by Andy2No** [reply]
You're welcome, @earlephilhower :)  Thanks for checking.

**Comment by Zitt** [comment]
The issue here was id 10 t error. 
I had selected the MBED OS version not the 
![correctbrd](https://user-images.githubusercontent.com/13250637/174406420-a367c43f-82f6-4d2a-bdca-bb929d68452f.png)
